tpm2/tpm2.go

```diff
@@ -1187,12 +1187,18 @@ func NVUndefineSpace(rw io.ReadWriter, ownerAuth string, owner, index tpmutil.Ha
 }
 
 // NVUndefineSpaceSpecial This command allows removal of a platform-created NV Index that has TPMA_NV_POLICY_DELETE SET.
-func NVUndefineSpaceSpecial(rw io.ReadWriter, index, platformPP tpmutil.Handle) error {
-	cmd, err := tpmutil.Pack(index, platformPP)
+func NVUndefineSpaceSpecial(rw io.ReadWriter, nvIndex, sessionIndex tpmutil.Handle, pAuth []byte) error {
+	indexAuth := AuthCommand{Session: sessionIndex, Attributes: AttrContinueSession, Auth: []byte(EmptyAuth)}
```

Comment:
There's a couple of things at play here,

- This function assumes that the auth value on the index is the empty auth, but that's not going to be universally true.
- UndefineSpaceSpecial requires authorization by policy that contains an explicit `PolicyCommandCode(CC_UndefineSpaceSpecial)`
- I think the solution is to take an AuthCommand for this and document the function to remind callers that they need to use a specially constructed policy to be allowed to perform this command.